### PR TITLE
Split out the /docs automated test section and enhance it a bit

### PR DESCRIPTION
**Pull request by camd**
Having the ``test`` and ``installation`` info interleaved didn't feel ideal to me.  Hopefully this organization will make it easier to get to the info you want.  :)

Please feel free to offer suggestions for more info!

### CODE REVIEW

**Comment by camd** [comment]
Thanks, @armenzg : Yeah, I forgot to add it to the TOC!  Done.

File: docs/testing.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
+# Running the Automated Tests
+
+## JavaScript
+
+### Validating JavaScript
```
Comment (sarah-clements) [inline docs/testing.md]: Do you we have info about the use of `yarn format` for prettier formatting? I forget about it sometimes until Travis yells at me.

Comment (camd) [reply docs/testing.md]: Good call.  Doesn't hurt to add it here, too.

**Comment by armenzg** [comment]
In production.

**Pull request merged**